### PR TITLE
Add voltage metric for server

### PR DESCRIPTION
**Pull request by Pengpengwanga**
Signed-off-by: Pengpeng Wang <pengpengw@vmware.com>

### CODE REVIEW

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 1]
```diff
@@ -149,8 +151,19 @@
             result.add(data);
             break;
          case MetricName.PDU_VOLTAGE:
-            data.setMetricName(String.format(MetricKeyName.SERVER_CONNECTED_PDUX_OUTLETX_VOLTAGE, pduAssetId, value.getExtraidentifier()));
-            result.add(data);
+            String extraidentifier = value.getExtraidentifier();
+            if(extraidentifier.contains(FlowgateConstant.OUTLET_NAME_PREFIX)) {
+               data.setMetricName(String.format(MetricKeyName.SERVER_CONNECTED_PDUX_OUTLETX_VOLTAGE, pduAssetId, value.getExtraidentifier()));
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:156]: check if the value is empty

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 2]
```diff
@@ -149,8 +151,19 @@
...
+               result.add(data);
+            }
+            time = data.getTimeStamp();
+            //some pdus without outlet metrics,but have inlet metrics
+            if(serverVoltage == null) {
+               serverVoltage = data.getValueNum();
+            }
+            if(extraidentifier.contains(FlowgateConstant.OUTLET_NAME_PREFIX)) {
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:164]: you already have check in line 155 why have another check outlet here ?

Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:160]: Removed

File: flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java [hunk 3]
```diff
@@ -127,6 +127,8 @@
 
    private List<MetricData> generateServerPduMetricData(List<ValueUnit> valueUnits, String pduAssetId){
       List<MetricData> result = new ArrayList<MetricData>();
+      Double serverVoltage = null;
+      long time = 0;
```
Comment (yixingjia) [inline flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java:131]: let' rename it to serverVoltageReadTime

Comment (Pengpengwanga) [reply flowgate-api/src/main/java/com/vmware/flowgate/service/AssetService.java]: Done

**Pull request merged**